### PR TITLE
Added build scripts for x86 and x64 build

### LINKED ISSUE #28: Windows x64 build
Hi,

I'm trying to build editorconfig/editorconfig-notepad-plus-plus#9 for x64.
Which build options should i set to build pcre?

Regards
Michael

### PR DESCRIPTION
**Pull request by ViceIce**
* Compile for `x86` and `x64`
* Test `utf_8_char` is failing on windows
* This should fix #28

### CODE REVIEW

**Comment by xuhdev** [comment]
Sorry I forgot to submit my review one week ago...

File: build.ps1 [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    if($arch -eq "x64"){
+        $gen += " Win64"
+    }
+
+    mkdir $dest -ErrorAction SilentlyContinue | Out-Null
+    Push-Location $dest
+
+    switch ($proj) {
+        pcre {
+            $BUILD_SHARED_LIBS = "OFF"
```
Comment (ViceIce) [inline build.ps1]: Ok. I had troubles with the tests, when i build pcre as shared library. I will look at it.

**Comment by ViceIce** [comment]
The pcre is now build as shared lib, if editor config is build as shared lib.

**Comment by xuhdev** [comment]
Oops, sorry, I guess I really missed you message... Can you give a short explanation on how to use these scripts?

**Comment by ViceIce** [comment]
First use the init.ps1 to fetch and unpack pcre, you can override the pcre version with the `-pcre` param.

Then you have to call build.ps1. The `-init` param (re)generates the vs build files for visual studio. The '-install` param is used to install all libs to the local `.\bin` directory.

`-static` can be **`ON`** or `OFF`
`-vsver` visual studio version (12,14 or **15**) to build for 
`-arch` can be `x86` or **`x64`**
`-proj` can be **`all`**, `pcre` or `core`

**Comment by ViceIce** [comment]
I can have a look on it, but not before 19.11.

**Comment by xuhdev** [comment]
No problem! Can you also enable `appveyor.yml` test for these scripts? Thanks for your contribution!

**Comment by ViceIce** [comment]
Ok

**Comment by ViceIce** [comment]
Done 😀

**Comment by xuhdev** [comment]
Thanks or your update. I test it, and got the following error when running `.\build.ps1`. Any ideas?

```
Missing build path! Used init?
At C:\Users\hong\Documents\src\editorconfig-core-c-master\build.ps1:120 char:5
+     throw "Missing build path! Used init?"
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Missing build path! Used init?:String) [], RuntimeException
    + FullyQualifiedErrorId : Missing build path! Used init?
```

**Comment by ViceIce** [comment]
Have you used the init parameter?

I think I should autogenerate the build scripts, when they are missing. 😉

**Comment by xuhdev** [comment]
Oh, that's the thing that I missed. Now I get this error

```
CMake Error at C:/Program Files/CMake/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PCRE2 (missing: PCRE2_LIBRARY PCRE2_INCLUDE_DIR)
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  CMake_Modules/FindPCRE2.cmake:27 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  src/CMakeLists.txt:30 (find_package)
```

**Comment by ViceIce** [comment]
I really have to optimize the script.

Please use the install parameter.

File: INSTALL.md [hunk 2]
```diff
@@ -89,9 +89,46 @@ PREFIX/lib as one of the library searching directory on UNIX/Linux to make sure
...
+
+```powershell
+~> ./build.ps1 [-proj all | pcre | core] [-init] [-install] [-vsver 15 | 14 | 12] [-arch x64 | x86] [-config Release | Debug]
+```
+
+Arguments:
+
+    -proj Optional, project to build.
+    -init Optional, (re)generate cmake build files, required first time or on `CMakeLists.txt` changes.
+    -install Optional, install to `bin/$ARCH(-static)/build` folder. Required for `pcre2` to build core.
```
Comment (xuhdev) [inline INSTALL.md]: I think the instruction on this line has confused me. What does it mean by "required for `pcre2` to build core"?

Comment (ViceIce) [reply INSTALL.md]: Ok, i extended the instructions. 😀

**Comment by xuhdev** [comment]
Are you satisfied with the current version? Just to confirm. It would be nice if you can add one line in Changelog (Add a new version 0.12.4).

**Comment by xuhdev** [comment]
Awesome, thanks!

**Pull request merged**